### ISSUE TITLE
code block custom ID is not respected when highlighting lines

### ISSUE THREAD

**Issue by hellt** [squidfunk/mkdocs-material#6448]
### Bug description

I am in love with the way one can link to certain line numbers from the markdown:

```
Here we [handle the `version`](#__codelineno-8-2:9) CLI flag.
```

The problem I am having is that `codelineno-8-2:9` is not a persisten id of the code block element.

If we add a code block before the one in question, its ID will auto-increment and the link won't work anymore.

Since pymarkdownx snippet extension allows to assign user-defined ID it is possible to have a static reference to the code block, that is agnostic to the document layout.

````
``` {.python #myid linenums="1"}
import bar
```
````

The problem is that `#myid` id assigned to the code block is not "respected" by the mkdocs-material, so that when one uses `#__codelineno-myid-7` the relevant line does not get highlighted, although the span elements have that ID.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
-

### Reproduction

[9.4.14+insiders.4.46.0-code-block-custom-id.zip](https://github.com/squidfunk/mkdocs-material/files/13598800/9.4.14%2Binsiders.4.46.0-code-block-custom-id.zip)

### Steps to reproduce

launch the reproduction, open `/blog/2023/11/11/bar/` page and click on the links at the bottom of the page.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and providing the reproduction. Fixed in [`39792ef`](https://github.com/squidfunk/mkdocs-material-insiders/commit/39792efd61ff73e8f0be1b9474492905d959cbf7).

**Comment by hellt** [comment]
Thanks @squidfunk!
